perception/autoware_map_based_prediction/src/map_based_prediction_node.cpp

```diff
@@ -1595,28 +1664,58 @@ LaneletsData MapBasedPredictionNode::getCurrentLanelets(const TrackedObject & ob
 
     LaneletsData object_lanelets;
     std::optional<std::pair<double, lanelet::Lanelet>> closest_lanelet{std::nullopt};
-    for (const auto & lanelet : surrounding_lanelets) {
+    for (size_t i = 0; i < surrounding_lanelets.size(); i++) {
       // Check if the close lanelets meet the necessary condition for start lanelets and
       // Check if similar lanelet is inside the object lanelet
-      if (!checkCloseLaneletCondition(lanelet, object) || isDuplicated(lanelet, object_lanelets)) {
+      if (
+        !checkCloseLaneletCondition(surrounding_lanelets[i], object) ||
+        isDuplicated(surrounding_lanelets[i], object_lanelets)) {
         continue;
       }
-
       // Memorize closest lanelet
       // NOTE: The object may be outside the lanelet.
-      if (!closest_lanelet || lanelet.first < closest_lanelet->first) {
-        closest_lanelet = lanelet;
+      if (!closest_lanelet || surrounding_lanelets[i].first < closest_lanelet->first) {
+        closest_lanelet = surrounding_lanelets[i];
       }
-
-      // Check if the obstacle is inside of this lanelet
       constexpr double epsilon = 1e-3;
-      if (lanelet.first < epsilon) {
-        const auto object_lanelet =
-          LaneletData{lanelet.second, calculateLocalLikelihood(lanelet.second, object)};
-        object_lanelets.push_back(object_lanelet);
+      for (size_t t = 0; t < surrounding_lanelets.size(); t++) {
```

Comment:
@YoshiRi I'm asking just to be sure. I think you marked the wrong part of the code. According to your second comment, I think you're talking about this loop, right?